polaris-service/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -248,11 +248,17 @@ public void initialize(String name, Map<String, String> properties) {
           ioImplClassName,
           storageConfigurationInfo);
     } else {
-      ioImplClassName = storageConfigurationInfo.getFileIoImplClassName();
-      LOGGER.debug(
-          "Resolved ioImplClassName {} for storageConfiguration {}",
-          ioImplClassName,
-          storageConfigurationInfo);
+      if (storageConfigurationInfo != null) {
+        ioImplClassName = storageConfigurationInfo.getFileIoImplClassName();
+        LOGGER.debug(
+            "Resolved ioImplClassName {} for storageConfiguration {}",
+            ioImplClassName,
+            storageConfigurationInfo);
+      } else {
+        throw new ValidationException(
```

Comment:
Should we just log a warning here since storageConfiguration could be null?

Reply:
@aihuaxu Are you requesting that no exception get thrown here?

Reply:
Yeah. What I meant is not to throw exception but just log a warning since storageConfiguration could be null.